packages/react/src/components/FileUploader/FileUploaderItem.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { settings } from 'carbon-components';
+import classNames from 'classnames';
+import { Filename } from './FileUploader';
+import { keys, matches } from '../../internal/keyboard';
+import uid from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderItem({
+  uuid,
+  name,
+  status,
+  iconDescription,
+  onDelete,
+  invalid,
+  errorSubject,
+  errorBody,
+  ...other
+}) {
+  const classes = classNames(`${prefix}--file__selected-file`, {
+    [`${prefix}--file__selected-file--invalid`]: invalid,
+  });
+  return (
+    <span className={classes} {...other}>
+      <p className={`${prefix}--file-filename`}>{name}</p>
+      <span className={`${prefix}--file__state-container`}>
+        <Filename
+          iconDescription={iconDescription}
+          status={status}
+          onKeyDown={evt => {
+            if (matches(evt, [keys.Enter, keys.Space])) {
+              if (status === 'edit') {
+                onDelete(evt, { uuid });
+              }
+            }
+          }}
+          onClick={evt => {
+            if (status === 'edit') {
+              onDelete(evt, { uuid });
+            }
+          }}
+        />
+      </span>
+      {invalid &&
+        (errorSubject && (
+          <div className={`${prefix}--form-requirement`}>
+            {errorSubject}
+            {errorBody && (
+              <p className={`${prefix}--form-requirement__supplement`}>
+                {errorBody}
+              </p>
+            )}
+          </div>
+        ))}
+    </span>
+  );
+}
+
+FileUploaderItem.propTypes = {
+  /**
+   * Unique identifier for the file object
+   */
+  uuid: PropTypes.string.isRequired,
```

Comment:
Do we need `uuid` or can we use file index in the `onDelete` handler?

Reply:
maybe but I'm just usually wary of using index as an identifier

Reply:
Did y'all reach a resolution on this particular issue? It seems to be the only outstanding issue on this review.

Reply:
@shinytoyrobots I think it's the same thing as the other PR you brought up the other day just in terms of size. Hard to find a solid chunk of time to go through it all 👍

Reply:
Got it. I do want to make sure that while we know we do want to split up PRs into more manageable sizes more proactively, let's also make sure we don't leave existing PRs open for too long. If we could find some time to resolve and close this PR, we can also take do a post mortem on a couple of these bigger PRs and see if there's a way we might have split them differently.

Reply:
The issue (on my end) is that it's hard to find a large enough time to review, so it's hard to reach a resolution without that time slot. Definitely not meaning to block this, but we did learn from that thread in Slack that sometimes folks will approve based on good faith due to size or parts of a codebase that it is touching.

The recommendation was that we could use PR review sessions as a way to educate and have folks feel comfortable about PRs, but the last few PR review sessions seemed to suggest that we'd need a deeper dive into areas of the codebase for folks to understand in order to review this work thoroughly.